lib/Cake/Test/Case/Model/Datasource/Database/MysqlTest.php

```diff
@@ -214,19 +214,19 @@ public function testTinyintCasting() {
 		$this->assertTrue((bool)$this->model->save(array('bool' => 5, 'tiny_int' => 5)));
 		$result = $this->model->find('first');
 		$this->assertTrue($result['Tinyint']['bool']);
-		$this->assertSame($result['Tinyint']['tiny_int'], '5');
+		$this->assertEquals($result['Tinyint']['tiny_int'], '5');
 		$this->model->deleteAll(true);
 
 		$this->assertTrue((bool)$this->model->save(array('bool' => 0, 'tiny_int' => 100)));
 		$result = $this->model->find('first');
 		$this->assertFalse($result['Tinyint']['bool']);
-		$this->assertSame($result['Tinyint']['tiny_int'], '100');
+		$this->assertEquals($result['Tinyint']['tiny_int'], '100');
 		$this->model->deleteAll(true);
 
 		$this->assertTrue((bool)$this->model->save(array('bool' => true, 'tiny_int' => 0)));
 		$result = $this->model->find('first');
 		$this->assertTrue($result['Tinyint']['bool']);
-		$this->assertSame($result['Tinyint']['tiny_int'], '0');
+		$this->assertEquals($result['Tinyint']['tiny_int'], '0');
```

Comment:
[memo]
This change addresses the following test failures.

```
2) MysqlTest::testTinyintCasting

Failed asserting that '5' is identical to 5.

/home/runner/work/cakephp2-php8/cakephp2-php8/lib/Cake/Test/Case/Model/Datasource/Database/MysqlTest.php:217
```

The following changes were made to the PDO MySQL/SQLite Driver in PHP 8.1
#### MySQL Driver
>Integers and floats in result sets will now be returned using native PHP types instead of strings when using emulated prepared statements.
>https://www.php.net/manual/en/migration81.incompatible.php#migration81.incompatible.pdo.mysql

#### SQLite Driver
>Integers and floats in results sets will now be returned using native PHP types.
>https://www.php.net/manual/en/migration81.incompatible.php#migration81.incompatible.pdo.sqlite

In `MysqlTest::testTinyintCasting()`, the value retrieved from the MySQL database was asserted with `assertSame()`, which seems to be a new failure in PHP 8.1.

* In PHP8.0 & MySQL tests, integer and float columns retrieved from the database were mapped to PHP strings
* In PHP8.1 & MySQL tests, those columns were mapped to PHP integer/float
Even with this type change, the tests in `ModelReadTest` and `ModelIntegrationTest` that assert records passed. (because of the use of `assertEquals()`)

Now, I thought about how to make changes in response to this failure.

`PDO::ATTR_STRINGIFY_FETCHES` can be set as `flags` in `database.php`.
That is, the PHP data type of a record retrieved from a database containing an integer/float column could be controlled by setting `database.php`.
In fact, can find an example of retrieving PHP int/float data with `PDO::ATTR_STRINGIFY_FETCHES => false`.
https://github.com/cakephp/cakephp/issues/1971

From this information, I thought it would be a good idea to use `assertEquals()` instead of `assertSame()` in the test, without changing the framework code.

Also, when someone upgrades their application to PHP8.1, I thought it would be nice to have some documentation left somewhere for those who want to maintain the column type behavior.

So I added a comment to `database.php.default`.